src/Controller/Services.php

```diff
@@ -71,6 +71,10 @@ public function processRequest(Request $request, RouteMatchInterface $route_matc
         $service_def->setContext($context_id, $context);
       }
     }
+
+    //Check access on the service definition.
+    $service_def->checkAccess($request, $route_match, $this->serializer);
```

Comment:
No, this work needs to be done in https://github.com/d8-contrib-modules/services/blob/8.x-4.x/src/Routing/ServiceEndpoint.php#L45 before we assign the route into the routes array.